src/mtcp/mtcp_restart.c

```diff
@@ -239,9 +239,11 @@ MTCP_PRINTF("Attach for debugging.");
       mtcp_abort();
     }
     // This assumes that the MTCP header signature is unique.
-    do {
-      mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
-    } while (mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
+    mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
```

Comment:
The problem with calling `mtcp_readfile` only _once_ is that it would always fail to find the `MTCP_SIGNATURE` because the first 4096 bytes (could be more) represent DMTCP metadata (i.e. ProcessInfo). Here is a test run of this patch:

```
## readdmtcp.sh on a compressed ckpt image:
dekaksi:~/dmtcp> util/readdmtcp.sh ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp 
*** ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp is a gzipped file. Will uncompress it into ckpt_tmp.dmcp first.
[20953] mtcp_restart.c:202 main:
  Considering 'ckpt_tmp.dmtcp' as a ckpt image.
[20953] mtcp_restart.c:244 main:
  ***ERROR: ckpt image doesn't match MTCP_SIGNATURE
Segmentation fault (core dumped)

## Uncompress the ckpt image:
dekaksi:~/dmtcp> mv ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp{,.gz}
‘ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp’ -> ‘ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp.gz’
dekaksi:~/dmtcp> gunzip ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp.gz 

## readmtcp.sh on uncompressed image
dekaksi:~/dmtcp> util/readdmtcp.sh ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp
[20992] mtcp_restart.c:202 main:
  Considering 'ckpt_sleep_1d4a852a5f139a6-40000-5511c7a1.dmtcp' as a ckpt image.
[20992] mtcp_restart.c:244 main:
  ***ERROR: ckpt image doesn't match MTCP_SIGNATURE
Segmentation fault (core dumped)
```